mage_ai/frontend/pages/triggers/index.tsx

```diff
@@ -37,10 +42,35 @@ function TriggerListPage() {
     [project?.features],
   );
 
+  // Debounce search input
```

Comment:
If you include the `Search` button, including the auto-fetch after user types in the keyboard with debounce is unnecessary. Use one or the other. You can just stick with fetching after the user explicitly hits "search" or presses "enter". Using the debounce with fetch after typing in the keyboard is prone to bugs or unexpected behavior, or users make typos / have different typing speeds, result in unnecessary requests. There are times to use that, but at least in this case, it's prob simpler to just remove that feature, which means you don't need this useEffect.